File: OrderManagementSystem/src/main/java/com/order/service/OrderService.java
package com.order.service;

import java.util.List;

import com.order.entity.Order;


public interface OrderService {

	public Order createOrder(Order order);

	public Order getOrder(int id);

	public List<Order> getAllOrder();
	
	public void deleteOrder(int id);
	
	public int updateOrder(Order order,int id);

}

File: Angular/orderManagement/src/app/service/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';


const BASE_URL="http://localhost:8089/";

@Injectable({
  providedIn: 'root'
})
export class OrderService {

  createUser(order:{id:number,orderName:string,orderPrice:number,brand:string}){
    return this.http.post(BASE_URL+'saveorder',order);
  }

  getOrder(){
    return this.http.get(BASE_URL+'findallorder');
  }

  deleteOrder(order:any){
    return this.http.delete(BASE_URL+'deleteorder/'+order.id);
  }
 getCurrentOrder(id){
  return this.http.get(BASE_URL+'findorder/'+id);
 }

 updateOrder(order:any){
  return this.http.put(BASE_URL+'updateorder/'+order.id,order);
 }

  constructor(public http:HttpClient) { }
}
